lbrynet/core/client/ConnectionManager.py

```diff
@@ -147,27 +152,27 @@ def protocol_disconnected(self, peer, protocol):
             del self._connections_closing[peer]
             d.callback(True)
 
+    def manage(self):
+        self._manage_deferred = self._manage()
+        if not self.stopped:
+            self._next_manage_call = self.callLater(self.MANAGE_CALL_INTERVAL_SEC, self.manage)
+
+
     @defer.inlineCallbacks
     def _manage(self):
-        self._manage_deferred = defer.Deferred()
-
-        from twisted.internet import reactor
         if len(self._peer_connections) < conf.settings['max_connections_per_stream']:
             log.debug("%s have %d connections, looking for %d",
                         self._get_log_name(), len(self._peer_connections),
                         conf.settings['max_connections_per_stream'])
+            ordered_request_creators = self._rank_request_creator_connections()
+            peers = yield self._get_new_peers(ordered_request_creators)
+            peer = self._pick_best_peer(peers)
```

Comment:
are you certain that this code won't throw an exception?

I'm pretty sure that an exception here will just disappear into the ether, as indicated in the comment
`# log this otherwise it will just end up as an unhandled error in deferred`